soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/SoQLError.scala

```diff
@@ -820,8 +820,17 @@ object SoQLAnalyzerError {
 
       implicit def encode[RNS: JsonEncode] = new SoQLErrorEncode[NoSuchColumn[RNS]] {
         override val code = tag
-        def encode(err: NoSuchColumn[RNS]) =
-          result(Fields(err.qualifier, err.name), s"No such column `${err.qualifier.fold("")("@" + _ + ".")}${err.name}'", err.source)
+        def encode(err: NoSuchColumn[RNS]) = {
+          val msg = new StringBuilder
+          msg.append("No such column `")
+            .append(err.qualifier.fold("")("@" + _ + "."))
+            .append(err.name)
+            .append("'")
+          if(err.qualifier.isEmpty) {
+            msg.append("  (Note: joined columns must be fully qualified)")
```

Comment:
There's an extra space here.
```suggestion
            msg.append(" (Note: joined columns must be fully qualified)")
```
If we care.